internal/datacoord/compaction_trigger.go

```diff
@@ -315,31 +318,44 @@ func (t *compactionTrigger) updateSegmentMaxSize(segments []*SegmentInfo) (bool,
 	collectionID := segments[0].GetCollectionID()
 	indexInfos := t.meta.GetIndexesForCollection(segments[0].GetCollectionID(), "")
 
-	isDiskANN := false
-	for _, indexInfo := range indexInfos {
-		indexType := getIndexType(indexInfo.IndexParams)
-		if indexType == indexparamcheck.IndexDISKANN {
-			// If index type is DiskANN, recalc segment max size here.
-			isDiskANN = true
-			newMaxRows, err := t.reCalcSegmentMaxNumOfRows(collectionID, true)
-			if err != nil {
-				return false, err
-			}
-			if len(segments) > 0 && int64(newMaxRows) != segments[0].GetMaxRowNum() {
-				log.Info("segment max rows recalculated for DiskANN collection",
-					zap.Int64("old max rows", segments[0].GetMaxRowNum()),
-					zap.Int64("new max rows", int64(newMaxRows)))
-				for _, segment := range segments {
-					segment.MaxRowNum = int64(newMaxRows)
-				}
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+	defer cancel()
+	collMeta, err := t.handler.GetCollection(ctx, collectionID)
+	if err != nil {
+		return false, fmt.Errorf("failed to get collection %d", collectionID)
+	}
+	vectorFields := typeutil.GetVectorFieldSchemas(collMeta.Schema)
+	fieldIndexTypes := lo.SliceToMap(indexInfos, func(t *model.Index) (int64, indexparamcheck.IndexType) {
+		return t.FieldID, getIndexType(t.IndexParams)
+	})
+	vectorFieldsWithDiskIndex := lo.Filter(vectorFields, func(field *schemapb.FieldSchema, _ int) bool {
+		if indexType, ok := fieldIndexTypes[field.FieldID]; ok {
+			return indexparamcheck.IsDiskIndex(indexType)
+		}
+		return false
+	})
+
+	allDiskIndex := len(vectorFields) == len(vectorFieldsWithDiskIndex)
+	if allDiskIndex {
```

Comment:
Maybe we should re-calculate the max segment size as long as there are any disk indexes.

Reply:
May result in larger memory index